### PR TITLE
pantheon.elementary-files: 4.5.0 -> 6.0.0

### PR DESCRIPTION
**Pull request by bobby285271**
###### Motivation for this change

See #121847 and https://github.com/elementary/files/releases/tag/6.0.0

~~Though [project README](https://github.com/elementary/files/tree/4429d7bc1f8a96c52bb329778bd96aff1135c55e) says libgranite-dev >= 6.0.0 will be okay to build this, but using `pantheon.granite` 6.0.0 will lead to the same issue in https://github.com/elementary/files/issues/1754, the app builds with `pantheon.granite` 6.1.0, so I guess #130357 needs to be merged before this one can be merged.~~

Now the apps requires `pantheon.granite` 6.1.0, so #130357 needs to be merged before this one can be merged. :-)

To be honest, I have no idea what the file `0001-filechooser-module-hardcode-gsettings-for-nixos.patch` does, but I will still keep part of its change in this PR. Please let me know if I did anything wrong.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130368` at 1726891a run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/506224b7ebcab30936c4cf788c046a5b">pantheon.elementary-files</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by jtojnar** [comment]
The reason we need the patch is because on NixOS GSettings schemas are not installed to `/usr` so applications need to load them from `XDG_DATA_DIRS`. But if some app loads the GTK module from `files` to replace the file chooser with Pantheon-specific one, it cannot just modify the app's environment so it would not be able to find the settings schemas and crash. See https://nixos.org/manual/nixpkgs/stable/#ssec-gnome-settings for more info.

**Comment by jtojnar** [comment]
One thing that we should probably do, in line with the changes, is add `elementary-files` to `xdg.portal.extraPortals` option in the module so that the file chooser works in Flatpak apps too.

https://github.com/NixOS/nixpkgs/blob/7dbc17410ce990c61d5a7a498380f80cddcc4f39/nixos/modules/services/x11/desktop-managers/pantheon.nix#L227

**Comment by bobby285271** [comment]
@jtojnar Please take a look :-)

**Comment by jtojnar** [comment]
Interesting, it does not seem like the GTK module is there any more. I wonder what the legacy dialog is used and if the patch is still necessary.

File: pkgs/desktops/pantheon/apps/elementary-files/default.nix [hunk 1]
```diff
@@ -73,22 +75,24 @@ stdenv.mkDerivation rec {
     libdbusmenu-gtk3
     libgee
     libgit2-glib
+    libhandy
     libnotify
     libunity
```
Comment (jtojnar) [inline pkgs/desktops/pantheon/apps/elementary-files/default.nix:80]: Does not seem to be used any more.

**Comment by jtojnar** [comment]
https://github.com/elementary/files/pull/1512#issuecomment-750410051

Looks like they use https://github.com/elementary/default-settings/pull/204 but later reverted it, opting to use Flatpak for apps https://github.com/elementary/default-settings/pull/215

That will not work for us.

**Comment by bobby285271** [comment]
> [elementary/files#1512 (comment)](https://github.com/elementary/files/pull/1512#issuecomment-750410051)
> 
> Looks like they use [elementary/default-settings#204](https://github.com/elementary/default-settings/pull/204) but later reverted it, opting to use Flatpak for apps [elementary/default-settings#215](https://github.com/elementary/default-settings/pull/215)
> 
> That will not work for us.

Shall I drop the last commit fd7bafcba7b59fc432a6f9c6792e3a8e2eff3d4e ?

**Comment by jtojnar** [comment]
No, we want that for Flatpak.

It would be also nice if we could use it for non-Flatpak apps but that would require us to add the GTK module (or patch GTK to use the portal even when `GTK_USE_PORTAL` is not set).

**Comment by bobby285271** [comment]
Putting this upgrade into #130380 for more testing as I found the old version of `elementary-files` doesn't respect some of the system settings (dark mode for example) in the new Pantheon desktop. Will still be great if anyone can kindly help me improve the upgrade if possible (to be honest I don't know what to do next currently)...

**Pull request closed without merging**